### DISCUSSION TITLE
prefetching dynamic import with ssr: false

### DISCUSSION THREAD

**Discussion by stefanprobst** [vercel/next.js#14058] [Help]
is it possible to somehow preload/prefetch a dynamic import that should only render on the client, i.e. needs to set `ssr: false`?

i realize that `next` will automatically preload dynamic imports when `ssr: true` and that preloading is [intentionally disabled](https://github.com/vercel/next.js/blob/canary/packages/next/next-server/lib/dynamic.tsx#L56-L63) otherwise, but maybe there is a way? my usecase is dynamically importing `netlify-cms-app`.

**Comment by stefanprobst** [comment]
a very hacky way seems to be with a custom loading component, but that's obviously not great
```js
{
  loading: () => {
    const register = React.useContext(LoadableContext);
    if (register) {
      register("netlify-cms-app");
    }
    return <div>Loading &hellip;</div>;
  },
  ssr: false,
}
```

**Reply by k4mr4n** [reply]
@stefanprobst You can simply run preload function. it worked for me on componentDidMount of my component perfectly

```js
const preload = () => {
  import('./your/component')
}
```
